### ISSUE TITLE
Highlight completely missing after update

### ISSUE THREAD

**Issue by nxuv** [yorickpeterse/nvim-pqf#12]
![image](https://github.com/yorickpeterse/nvim-pqf/assets/60989682/3896e5dd-ca49-44a2-a13b-853777419ff2)

Tried removing pqf completely, starting without configs and colorscheme definitions but nothing worked

**Comment by nxuv** [comment]
![image](https://github.com/yorickpeterse/nvim-pqf/assets/60989682/e4216899-d62f-4248-aa80-f5ec56cbed3b)

strangely when using `cexpr system("command")` pqf works as supposed to, but not when using `setqflist` or `caddexpr`

**Comment by nxuv** [comment]
Going to be rolling to previous commit for now.

For reference, I've wrote custom builder using Plenary which adds line each output.

Also tested just now to close and reopen quickfix with `cclose` and `copen` and it did highlight filepath and aligned text, but signs still missing colour

**Comment by yorickpeterse** [comment]
@al1-ce 

- What version of NeoVim are you using? 
- What are your highlight settings for the groups listed at https://github.com/yorickpeterse/nvim-pqf#tweaking-the-highlights?
- Can you provide a way to reproduce this using a simple `call setqflist(...)` expression? Without that it's going to be hard to figure out what's going on

**Comment by nxuv** [comment]
Output of `hi GROUP_NAME`:
```
Directory       xxx ctermfg=12 guifg=#83a598 
Number          xxx ctermfg=13 guifg=#d3869b 
DiagnosticError xxx ctermfg=9  guifg=#fb4934 
DiagnosticWarn  xxx ctermfg=11 guifg=#fe8019 
DiagnosticInfo  xxx ctermfg=12 guifg=#83a598 
DiagnosticHint  xxx ctermfg=14 guifg=#8ec07c 
```

Set with:
```lua
local function highlightAll(names, val)
    for i = 1, #names do
        vim.api.nvim_set_hl(0, names[i], val)
    end
end

local style_function    = {fg = g.terminal_color_12, ctermfg = 12}
local style_keyword     = {fg = g.terminal_color_9, ctermfg = 9}
local style_const       = {fg = g.terminal_color_13, ctermfg = 13}
local style_macro       = {fg = g.terminal_color_14, ctermfg = 14}
local style_special     = {fg = "#fe8019", ctermfg = 11}

highlightAll({"Constant", "Boolean", "Number", "Float"}, style_const)
highlightAll({"Error", "DiagnosticError"}, style_keyword)
highlightAll({"Directory", "DiagnosticInfo"}, style_function)
highlightAll({"DiagnosticWarn"}, style_special)
highlightAll({"DiagnosticHint"}, style_macro)
```

For reproduction:
```vim
caddexpr("test.d(20): Error: test error")
```
and
```vim
call setqflist([{"filename": "test.t", "lnum": 10, "col": 1, "type": "E", "text": "Test"}], "a")
```
For caddexpr no highlight will be present at all

For setqflist repeat more then one time and it'll loose highlight even if values change
![image](https://github.com/yorickpeterse/nvim-pqf/assets/60989682/3006e1df-bc14-40f5-a475-c9e274561700)

Error format for this particulat error is 
```lua
vim.opt.errorformat:append("%f(%l): %t%*[^:]: %m")
```

And here's it working on older commit
![image](https://github.com/yorickpeterse/nvim-pqf/assets/60989682/5bcd6c09-7f2a-4bda-8e6c-9635231b6a4c)

**Comment by yorickpeterse** [comment]
@al1-ce I'm unable to reproduce this using your code example. For example, this is the output of `caddexpr` and the `call`:

![image](https://github.com/yorickpeterse/nvim-pqf/assets/86065/9e272f45-79ef-470a-ab79-52e184852d76)

Also, your screenshot doesn't seem to match what the code snippet does: it only adds two entries, yet you have more.

**Comment by yorickpeterse** [comment]
Oh I see now what you mean: if you call `setqflist` multiple times, the highlights get messed up.

**Comment by nxuv** [comment]
> For setqflist repeat more then one time and it'll loose highlight even if values change 

You missed one if my sentences 

Anyway, https://github.com/al1-ce/MonolithVim here's my full neovim config, tho it's on old version if pqf (changeable in plugins.lua)

I really dunno what's wrong and why it's not working

**Comment by nxuv** [comment]
Yea, actually I forgot to test it with `caddexpr`

**Comment by nxuv** [comment]
https://github.com/al1-ce/MonolithVim/blob/f3d6961a60c67655449aba0ea6c9a1400d2e08bd/lua/monolith/plugins/utils/justbuild.ts#L366

If you look at function at this line (can't do multiselect on mobile rn) you can see that I'm doing `caddexpr` many times as plenary spits out what job does. It's basically a backbone of my "make/build" system

**Comment by yorickpeterse** [comment]
@al1-ce 2dbdde286c2de754b986bcb2996c855225eb461a should fix this problem. Please give it a try and let me know if it fixes it for you as well.

**Comment by nxuv** [comment]
Seems to be working, but signs (i.e (X) on left of error) are still colourless

I am certainly defining them here:

https://github.com/al1-ce/MonolithVim/blob/f3d6961a60c67655449aba0ea6c9a1400d2e08bd/lua/monolith/core/colorscheme.lua#L265-L273

Have no idea what it might be

**Comment by yorickpeterse** [comment]
Using this:

```vim
call setqflist([{"filename": "test.t", "lnum": 10, "col": 1, "type": "E", "text": "Test"}], "a")
call setqflist([{"filename": "test.t", "lnum": 10, "col": 1, "type": "W", "text": "Test"}], "a")
call setqflist([{"filename": "test.t", "lnum": 10, "col": 1, "type": "N", "text": "Test"}], "a")
call setqflist([{"filename": "test.t", "lnum": 10, "col": 1, "type": "I", "text": "Test"}], "a")
```

Signs appear to be given the correct highlights for me:

![image](https://github.com/yorickpeterse/nvim-pqf/assets/86065/fe232fff-90a7-46b7-9827-eef7944ad090)

The highlight may not be applied if the `type` field of the quickfix item isn't recognized, i.e. it's not `E`, `W`, `I` or `N`. If you're supplying different values then that may explain the issue.

@al1-ce Please provide a standalone way to reproduce this. I can't reproduce this myself, and I'm not going to dig through an entire NeoVim configuration just to figure out what's going on.

**Comment by nxuv** [comment]
Tested and it's because this for some reason:
https://github.com/al1-ce/MonolithVim/blob/f3d6961a60c67655449aba0ea6c9a1400d2e08bd/lua/monolith/plugins/formatting/pqf.lua#L1-L7

I.e with custom "signs"

**Comment by yorickpeterse** [comment]
@al1-ce That should now be fixed in ba0e291cb5c0249838132e197dac082777bdc7d6. I think this should solve all remaining issues?

**Comment by nxuv** [comment]
Incredible, thank you
